esphome/components/image/__init__.py

```diff
@@ -74,6 +77,56 @@ def download_mdi(value):
     return value
 
 
+def get_filename_from_url(url):
+    # Regular expression pattern to match the file name at the end of the URL
+    pattern = r"/([^/]+)$"
+    match = re.search(pattern, url)
+    if match:
+        file_name = match.group(1)
```

Comment:
For consistency with the method's name (and the rest of the code), you might want to rename "file_name" to a single word filename (valid for all instances)
```suggestion
        filename = match.group(1)
```